main/cmds.go

```diff
@@ -72,44 +72,40 @@ func install(ctx *log.Context, h HandlerEnvironment, seqNum int) (string, error)
 	// If it is (2) attempt to migrate to mrseq.  Find the latest settings
 	// file, and set the sequence to it.
 	if _, err := os.Stat(mostRecentSequence); os.IsNotExist(err) {
-		migrateToMostRecentSequence(ctx, h, seqNum)
+		err := migrateToMostRecentSequence(ctx, seqNum)
+		if err != nil {
+			return "install failed", err
+		}
 	}
 
 	ctx.Log("event", "created data dir", "path", dataDir)
 	ctx.Log("event", "installed")
 	return "", nil
 }
 
-func migrateToMostRecentSequence(ctx *log.Context, h HandlerEnvironment, seqNum int) {
-	// The status folder is used instead of the settings because the settings file is written
-	// by the agent before install is called.  As a result, the extension cannot determine if this
-	// is a new install or an upgrade.
-	//
-	// If this is an upgrade there will be a status file. The agent will re-write the last status
-	// file to indicate that the upgrade happened successfully. The extension uses the last status
-	// sequence number to determine the last settings file that was executed.
-	//
-	// The agent helpfully copies mrseq every time an extension is upgraded thereby preserving the
-	// most recent executed sequence. If extensions use mrseq they benefit from this mechanism, and
-	// do not have invent another method.  The CustomScript extension should have been using this
-	// from the beginning, but it was not.
-	//
-	computedSeqNum, err := FindSeqNumStatus(h.HandlerEnvironment.StatusFolder)
+func migrateToMostRecentSequence(ctx *log.Context, seqNum int) error {
+	// The configSequenceNumber (seqNum) env variable is used for mrseq if mrseq not found
+	// based on logic in main.go the seqNum should always be populated
+	ctx.Log("event", "recreate mrseq", "message", fmt.Sprintf("cannot find mrseq at %v", mostRecentSequence))
+	fout, err := os.Create(mostRecentSequence)
+	defer fout.Close()
```

Comment:
We shouldn't defer fout.Close() if there's an error, so this should probably be moved below the error check